test/optimizer/test_statement_to_opr_convertor.py

```diff
@@ -346,6 +347,48 @@ def test_visit_select_limit(self, mock_p, mock_c, mock_d):
             wrong_plan.append_child(plan)
         self.assertNotEqual(wrong_plan, actual_plan)
 
+    @patch('src.optimizer.statement_to_opr_convertor.bind_dataset')
+    @patch('src.optimizer.statement_to_opr_convertor.bind_columns_expr')
+    @patch('src.optimizer.statement_to_opr_convertor.bind_predicate_expr')
+    def test_visit_select_sample(self, mock_p, mock_c, mock_d):
+        m = MagicMock()
+        mock_p.return_value = mock_c.return_value = mock_d.return_value = m
+        stmt = Parser().parse(""" SELECT data, id FROM video SAMPLE 2 \
+                   WHERE data > 2 LIMIT 3;""")[0]
+
+        converter = StatementToPlanConvertor()
+        actual_plan = converter.visit(stmt)
+        plans = []
+
+        plans.append(LogicalLimit(ConstantValueExpression(3)))
+
+        plans.append(LogicalProject(
+            [TupleValueExpression('data'), TupleValueExpression('id')]))
+
+        plans.append(
+            LogicalFilter(
+                ComparisonExpression(
+                    ExpressionType.COMPARE_GREATER,
+                    TupleValueExpression('data'),
+                    ConstantValueExpression(2))))
+
+        plans.append(LogicalSample(ConstantValueExpression(2)))
+
+        plans.append(LogicalGet(TableRef(TableInfo('video')), m))
+
+        expected_plan = None
+        for plan in reversed(plans):
+            if expected_plan:
+                plan.append_child(expected_plan)
+            expected_plan = plan
+
+        self.assertEqual(expected_plan, actual_plan)
+
+        wrong_plan = plans[0]
+        for plan in plans[1:]:
+            wrong_plan.append_child(plan)
+        self.assertNotEqual(wrong_plan, actual_plan)
```

Comment:
what is this wrong_plan?

Reply:
This is something that was in the other tests. It tests whether the plan has changed after adding commands I think.

Reply:
@gaurav274 What is the purpose for this wrong_plan? I am a little confused.

Reply:
Ahh, after thinking a lot, I recall adding it for one of the test cases, to double-check that the plan tree comparison is working correctly. Earlier it returned true though the plans were different. I don't think we need it for other.